src/main/java/com/epam/javacourses2016/task12/SolverTask12.java

```diff
@@ -1,22 +1,35 @@
 package com.epam.javacourses2016.task12;
 
+import java.util.Collections;
 import java.util.List;
 
 /**
  * Задан список целых чисел и число X.
  * Не используя вспомогательных объектов и не изменяя размера списка,
- *    переставить элементы списка так, чтобы сначала шли числа, не превосходящие X, а затем числа, большие X.
+ * переставить элементы списка так, чтобы сначала шли числа, не превосходящие X, а затем числа, большие X.
  */
 public class SolverTask12 {
 
     /**
      * Преобразует целочисленный список таким образом, чтобы сначала шли числа меньшие value, затем большие.
+     *
      * @param integers Целочисленный список.
-     * @param value Разделительное значение.
+     * @param value    Разделительное значение.
      * @return Преобразованный список.
      */
     List<Integer> transform(List<Integer> integers, int value) {
-        //TODO
-        return null;
+        boolean flag = true;
+        while (flag) {
+            flag = false;
+            for (int j = 0; j < integers.size() - 1; j++) {
+                if (integers.get(j) > integers.get(j + 1)) {
+                    if (integers.get(j) >= value) {
+                        Collections.swap(integers, j, j + 1);
+                        flag = true;
+                    }
+                }
+            }
+        }
+        return integers;
```

Comment:
I agree with you, but on the instructions: "Do not use the auxiliary objects and without changing the size of the list." It would be simpler to use Collections.sort. But I thought that the Task would have lost its meaning.